library/src/test/java/com/novoda/downloadmanager/lib/DownloadsStatusTest.java

```diff
@@ -0,0 +1,127 @@
+package com.novoda.downloadmanager.lib;
+
+import android.app.DownloadManager;
+
+import org.junit.Test;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+public class DownloadsStatusTest {
+
+    @Test
+    public void whenVisibilityIsCompletedThenNotificationIsToBeDisplayed() {
+        boolean toBeDisplayed = DownloadsStatus.isNotificationToBeDisplayed(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED);
+        assertThat(toBeDisplayed).isTrue();
+    }
+
+    @Test
+    public void whenVisibilityIsOnlyCompletionThenNotificationIsToBeDisplayed() {
+        boolean toBeDisplayed = DownloadsStatus.isNotificationToBeDisplayed(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_ONLY_COMPLETION);
+        assertThat(toBeDisplayed).isTrue();
+    }
+
+    @Test
+    public void whenVisibilityIsHiddenThenNotificationIsNotBeDisplayed() {
+        boolean toBeDisplayed = DownloadsStatus.isNotificationToBeDisplayed(DownloadManager.Request.VISIBILITY_HIDDEN);
+        assertThat(toBeDisplayed).isFalse();
+    }
+
+    @Test
+    public void whenStatusIs490ThenStatusIsCancelled() {
+        boolean isCancelled = DownloadsStatus.isStatusCancelled(490);
+        assertThat(isCancelled).isTrue();
+    }
+
+    @Test
+    public void whenStatusIs4xxThenStatusIsClientError() {
+        for (int i = 400; i < 500; i++) {
+            assertThat(DownloadsStatus.isStatusClientError(i)).isTrue();
+        }
+
+        assertThat(DownloadsStatus.isStatusClientError(399)).isFalse();
```

Comment:
The thing is that you cannot actually test all the values that are out of range, this is why I've put the limits inside the 4xx test instead of creating a test only for 399 and 500 considering that this test covers all the out of range values, when it does not